geode-apis-compatible-with-redis/src/main/java/org/apache/geode/redis/internal/data/RedisStringCommandsFunctionExecutor.java

```diff
@@ -144,6 +143,26 @@ public int setbit(RedisKey key, long offset, int value) {
         () -> getRedisString(key, false).setbit(getRegion(), key, value, byteIndex, bitIndex));
   }
 
+  @Override
+  public Void mset(List<RedisKey> keys, List<byte[]> values) {
+    List<RedisKey> keysToLock = new ArrayList<>(keys.size());
+    for (RedisKey key : keys) {
+      getRegionProvider().ensureKeyIsLocal(key);
+      keysToLock.add(key);
+    }
+
+    // Pass a key in so that the bucket will be locked. Since all keys are already guaranteed to be
+    // in the same bucket we can use any key for this.
+    return stripedExecuteInTransaction(keysToLock.get(0), keysToLock, () -> mset0(keys, values));
+  }
+
+  private Void mset0(List<RedisKey> keys, List<byte[]> values) {
+    for (int i = 0; i < keys.size(); i++) {
+      setRedisString(getRegionProvider(), keys.get(i), values.get(i));
```

Comment:
do you need to also clear any pending expiration in the case of mset being called on an existing key that has expiration scheduled? I think the easiest way to do this is call the "set" method with SetOptions param set to "null".

Reply:
Yes! Good call.